src/pilot/ovs_dpdk_setup.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2015-2017 Dell Inc. or its subsidiaries.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import logging
+import argparse
+import subprocess
+from ironicclient import client
+
+from credential_helper import CredentialHelper
+from logging_helper import LoggingHelper
+
+logging.basicConfig()
+LOG = logging.getLogger(os.path.splitext(os.path.basename(sys.argv[0]))[0])
+
+
+def get_ironic_client():
+    try:
+        os_auth_url, os_tenant_name, os_username, os_password = \
+            CredentialHelper.get_undercloud_creds()
+        kwargs = {
+            'os_username': os_username,
+            'os_password': os_password,
+            'os_auth_url': os_auth_url,
+            'os_project_name': os_tenant_name
+            }
+
+        ironic = client.get_client(1, **kwargs)
+    except:
+        raise Exception('ERROR: Error setting up ironic client.')
+    LOG.info("Ironic client authnetication successful.")
+    return ironic
+
+
+def get_ironic_nodes(ironic_client):
+    try:
+        nodes = ironic_client.node.list(detail=True)
+    except:
+        raise Exception('ERROR: Error retrieving ironic nodes.')
+    LOG.info("Retreiving baremetal nodes information from Ironic client.")
+    return nodes
+
+
+def setup_ovs_dpdk_environment(nics, config_file, env_file, mode):
+    assert os.path.isfile(
+        config_file), config_file + " file does not exist"
+    assert os.path.isfile(
+        env_file), env_file + " file does not exist"
+    if mode == 2:
+        interfaces = "'" + ",".join(nics[0:2]) + "'"
+    else:
+        interfaces = "'" + ",".join(nics) + "'"
+
+    cmds = [
+        'sed -i "s|changeme1|' + nics[0] + '|" ' + config_file,
+        'sed -i "s|changeme2|' + nics[1] + '|" ' + config_file,
+        'sed -i "s|changeme3|' + nics[2] + '|" ' + config_file,
+        'sed -i "s|changeme4|' + nics[3] + '|" ' + config_file,
+        'sed -i "s|Compute::Net::SoftwareConfig:.*|' +
+        'Compute::Net::SoftwareConfig: ' + config_file + '|" ' + env_file,
+        'sed -i "s|DpdkInterfaces:.*|DpdkInterfaces: ' +

```

Comment:
The line above and following, we need to keep.  Instead of using this mechanism, it seems like we should do something similar to how we modify dell-environment.yaml in automation.